src/lib/live-backup-copy.ts

```diff
@@ -0,0 +1,186 @@
+		  } )
+		: undefined;
+
+	const result = await api.mutate<
+		StartLiveBackupCopyMutation,
+		StartLiveBackupCopyMutationVariables
+	>( {
+		mutation: START_LIVE_COPY_MUTATION,
+		variables: {
+			input: {
+				id: appId,
+				tool: SQLDumpTool.MYSQLDUMP,
+				environmentId,
+				tables,
+				type: config.type,
+				subsiteIds: config.subsite_ids,
+				wpcliCommand: config.wpcli_command,
+			},
+		},
+	} );
+
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+	if ( ! result.data?.startLiveBackupCopy.copyId ) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+		throw new Error(
+			`Failed to start live backup copy: ${
+				result.data?.startLiveBackupCopy?.message
+					? result.data?.startLiveBackupCopy?.message
+					: 'Unkown error'
+			}`
+		);
+	}
+
+	return result.data?.startLiveBackupCopy.copyId;
+}
+
+export async function getDownloadURL( {
+	appId,
+	environmentId,
+	copyId,
+	timeoutInSeconds = 60 * 60,
+}: {
+	appId: number;
+	environmentId: number;
+	copyId: string;
+	timeoutInSeconds?: number;
+} ): Promise< string > {
+	const startTime = Date.now();
+	const timeoutMs = timeoutInSeconds * 1000;
+
+	const api = API( { exitOnError: true } );
+
+	while ( Date.now() - startTime < timeoutMs ) {
+		// eslint-disable-next-line no-await-in-loop
+		const result = await api.mutate<
+			GenerateLiveBackupCopyDownloadUrlMutation,
+			GenerateLiveBackupCopyDownloadUrlMutationVariables
+		>( {
+			mutation: GENERATE_LIVE_BACKUP_DOWNLOAD_URL_MUTATION,
+			variables: {
+				input: {
+					id: appId,
+					environmentId,
+					copyId,
+				},
+			},
+		} );
+
+		if (
+			result.data?.generateLiveBackupCopyDownloadURL?.url &&
+			! result.data?.generateLiveBackupCopyDownloadURL?.processing
+		) {
+			return result.data.generateLiveBackupCopyDownloadURL.url;
+		}
+
+		if ( ! result.data?.generateLiveBackupCopyDownloadURL?.success ) {
+			throw new Error(
+				`Failed to generate download URL: ${
+					result.data?.generateLiveBackupCopyDownloadURL?.url
+						? result.data?.generateLiveBackupCopyDownloadURL?.url
+						: 'Unknown error'
+				}`
+			);
+		}
+
+		if ( result.data?.generateLiveBackupCopyDownloadURL?.processing === true ) {
+			await new Promise( resolve => setTimeout( resolve, 5000 ) );
+		}
+	}
+
+	throw new Error( `Timeout: Download URL not generated within ${ timeoutInSeconds } seconds` );
+}
+
+export async function downloadFile( url: string, filename: string ) {
+	const file = fs.createWriteStream( filename );
+
+	return new Promise< string >( ( resolve, reject ) => {
+		https.get( url, response => {
+			response.pipe( file );
+
+			file.on( 'finish', () => {
+				file.close();
+				resolve( path.resolve( file.path as string ) );
+			} );
+
+			file.on( 'error', err => {
+				// TODO: fs.unlink runs in the background so there's a chance that the app dies before it finishes.
+				//  This needs fixing.
+				fs.unlink( filename, () => null );
+				reject( err );
```

Comment:
```suggestion
				try {
					fs.unlink( filename, () => null );
				} catch {
				    // Maybe log an error, if it makes sense
				}
				reject( err );
```

Reply:
Good point. I've changed it to use the node native fetch: https://github.com/Automattic/vip-cli/pull/2476/commits/043825efcbd88f4b5c78140de68358e46d120b6b

So, once we migrate everything to native node fetch, it will be one less place to fix. What do you think?